task6__israel_jacob/src/Cities.cpp

```diff
@@ -0,0 +1,161 @@
+}
+
+// Helper method to filter cities within a radius using the Euclidean distance
+std::vector<std::pair<std::string, Location>> Cities::normalEuclideanDistance(std::vector<std::pair<std::string, Location>> vCities, double radius, double x, double y) {
+	vCities.erase(std::remove_if(vCities.begin(), vCities.end(),
+		[x, y, radius](std::pair<std::string, Location> pair) {
+			double dx = pair.second.getX() - x;
+	double dy = pair.second.getY() - y;
+	return (dx * dx + dy * dy) >= radius * radius; }), vCities.end());
+	std::sort(vCities.begin(), vCities.end(),
+		[x, y](std::pair<std::string, Location> pair1, std::pair<std::string, Location> pair2) {
+			double dx1 = pair1.second.getX() - x;
+			double dy1 = pair1.second.getY() - y;
+			double dx2 = pair2.second.getX() - x;
+			double dy2 = pair2.second.getY() - y;
+			return dx1 * dx1 + dy1 * dy1 <= dx2 * dx2 + dy2 * dy2; });
+	return vCities;
+}
+
+// Helper method to filter cities within a radius using the infinity norm
+std::vector<std::pair<std::string, Location>> Cities::infinityNorm(std::vector<std::pair<std::string, Location>> vCities, double radius, double x, double y) {
+	vCities.erase(std::remove_if(vCities.begin(), vCities.end(),
+		[x, y, radius](std::pair<std::string, Location> pair) {
+			double dx = std::abs(pair.second.getX() - x);
+	double dy = std::abs(pair.second.getY() - y);
+	return std::max(dx, dy) >= radius;
+		}), vCities.end());
+	std::sort(vCities.begin(), vCities.end(),
+		[x, y](std::pair<std::string, Location> pair1, std::pair<std::string, Location> pair2) {
+			double dx1 = std::abs(pair1.second.getX() - x);
+	double dy1 = std::abs(pair1.second.getY() - y);
+	double dx2 = std::abs(pair2.second.getX() - x);
+	double dy2 = std::abs(pair2.second.getY() - y);
+	return std::max(dx1, dy1) <= std::max(dx2, dy2);
+		});
+	return vCities;
+}
+
+// Helper method to filter cities within a radius using the Manhattan distance
+std::vector<std::pair<std::string, Location>> Cities::manhattanDistance(std::vector<std::pair<std::string, Location>> vCities, double radius, double x, double y) {
+	vCities.erase(std::remove_if(vCities.begin(), vCities.end(),
+		[x, y, radius](std::pair<std::string, Location> pair) {
+			double dx = std::abs(pair.second.getX() - x);
+	double dy = std::abs(pair.second.getY() - y);
+	return dx + dy >= radius;
+		}), vCities.end());
+	std::sort(vCities.begin(), vCities.end(),
+		[x, y](std::pair<std::string, Location> pair1, std::pair<std::string, Location> pair2) {
+			double dx1 = std::abs(pair1.second.getX() - x);
+	double dy1 = std::abs(pair1.second.getY() - y);
+	double dx2 = std::abs(pair2.second.getX() - x);
+	double dy2 = std::abs(pair2.second.getY() - y);
+	return dx1 + dy1 <= dx2 + dy2;
+		});
+	return vCities;
+}
+
+// Method to determine the number of cities located north of a given city
+int Cities::northOfTheCity(std::vector<std::pair<std::string, Location>> vCities, std::string name)
+{
+	double y = cities.find(name)->second.getY();
```

Comment:
you are making the same calculations again, you should have passed the results and then to filter the cities of the north

Reply:
i do the calculate just once